src/core/NotePlayHandle.cpp

```diff
@@ -670,7 +670,9 @@ void NotePlayHandleManager::extend( int c )
 
 void NotePlayHandleManager::free()
 {
+	delete s_available[0];
```

Comment:
@sakertooth This should have been `std::free` since it's allocated with `malloc` for some reason, not `new`.

Additionally, I don't think freeing just the first element will fix all the memory leaks. If I'm reading it correctly, when `extend` is called, it allocates another array which starts at `s_availableIndex + 1` but that information is lost when `s_availableIndex` is updated which makes it impossible to deallocate that array correctly.

`extend` also leaves the memory from 0 to `s_availableIndex` uninitialized instead of copying the `NotePlayHandle`s from the old `s_available` array to the newly allocated one, which I believe causes a memory leak. And now you're trying to free that uninitialized memory at `s_available[0]` which would likely crash LMMS.

The `NotePlayHandleManager` is a convoluted mess that requires a refactor. That is why I suggested rewriting it to use `std::vector` instead - it would make the `NotePlayHandleManager` implementation easier and less error-prone, even if we don't want to keep `NotePlayHandleManager` in the long term.

Reply:
The `NotePlayHandle` destructors may need to be manually called. If it isn't doing that, that could also cause a memory leak.

Reply:
It is in `NotePlayHandleManager::release`. I the code currently seems to call that function accordingly when it needs to.

Reply:
> Unfortunately, I have little time to work on this.

Ok, then please revert this PR until we get a proper refactor. Better to have a status quo with memory leaks than a new regression that segfaults.

Reply:
On master, when closing LMMS, ASan show this:

```cpp
==31805==ERROR: AddressSanitizer: SEGV on unknown address (pc 0x5e57f0886166 bp 0x7fff1ce4df50 sp 0x7fff1ce4df30 T0)
==31805==The signal is caused by a READ memory access.
==31805==Hint: this fault was caused by a dereference of a high value address (see register values below).  Disassemble the provided pc to learn which register was used.
    #0 0x5e57f0886166 in lmms::NotePlayHandleManager::free() /home/saker/Documents/Git/lmms/src/core/NotePlayHandle.cpp:673:2
    #1 0x5e57f0e589fc in main /home/saker/Documents/Git/lmms/src/core/main.cpp:1017:2
    #2 0x763783239c87  (/usr/lib/libc.so.6+0x25c87) (BuildId: 32a656aa5562eece8c59a585f5eacd6cf5e2307b)
    #3 0x763783239d4b in __libc_start_main (/usr/lib/libc.so.6+0x25d4b) (BuildId: 32a656aa5562eece8c59a585f5eacd6cf5e2307b)
    #4 0x5e57f052f934 in _start (/home/saker/Documents/Git/lmms/build/lmms+0x582934) (BuildId: 42186b1de30ff6addd79fb8a31156cf1ddcd1e5c)
```

So `NotePlayHandleManager::free` is only called when the app is getting ready to close, so it doesn't seem to affect the user directly, but this still is a segmentation fault. We should just fix it once and for all until there's time for a better replacement. I'm okay with that idea.

Reply:
> The code currently seems to call that function accordingly when it needs to.

Yes, but it doesn't do so in `free` before deallocating the memory those objects were constructed in.

Reply:
Okay, fair, I'm reverting this.